Build with cuda 12.8 fails

**Describe the bug**
Compilation issues with cuda 12.8.1

**Platform (please complete the following information):**
- Machine: LLNL Matrix (H100)
- Compiler: gcc-12.1.1 + nvcc 12.8.1
- GEOS Version: c2768e6ecd648b17760afa187c44d5a7009d6b72 (develop as of 06/04/25)

**Additional context**
First error:
```
[  9%] Building CUDA object coreComponents/constitutive/CMakeFiles/constitutive.dir/fluid/multifluid/compositional/models/ImmiscibleWaterFlashModel.cpp.o
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp: In static member function ‘static void geos::constitutive::compositional::NegativeTwoPhaseFlash::computeDerivatives(geos::integer, geos::real64, geos::real64, geos::arraySlice1d<const double>&, const geos::constitutive::compositional::ComponentProperties::KernelWrapper&, const geos::constitutive::compositional::FlashData&, const geos::real64&, geos::arraySlice1d<const double, USD>&, geos::arraySlice1d<const double, USD>&, geos::arraySlice1d<double, USD2>&, geos::arraySlice2d<double, USD3>&, geos::arraySlice2d<double, USD3>&)’:
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:383:134: error: template argument 2 is invalid
  383 |     stackArray2d< real64, maxNumComps * maxNumDofs > logLiquidFugacityDerivs( numComps, numDofs );
      |                                                                                                                                      ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:383:178: error: expression list treated as compound expression in initializer [-fpermissive]
  383 |     stackArray2d< real64, maxNumComps * maxNumDofs > logLiquidFugacityDerivs( numComps, numDofs );
      |                                                                                                                                                                                  ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:384:134: error: template argument 2 is invalid
  384 |     stackArray2d< real64, maxNumComps * maxNumDofs > logVapourFugacityDerivs( numComps, numDofs );
      |                                                                                                                                      ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:384:178: error: expression list treated as compound expression in initializer [-fpermissive]
  384 |     stackArray2d< real64, maxNumComps * maxNumDofs > logVapourFugacityDerivs( numComps, numDofs );
      |                                                                                                                                                                                  ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:403:214: error: request for member ‘toSlice’ in ‘logLiquidFugacityDerivs’, which is of non-class type ‘int’
  403 |     FugacityCalculator::computeLogFugacityDerivatives( numComps,
      |                                                                                                                                                                                                                      ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:412:214: error: request for member ‘toSlice’ in ‘logVapourFugacityDerivs’, which is of non-class type ‘int’
  412 |     FugacityCalculator::computeLogFugacityDerivatives( numComps,
      |                                                                                                                                                                                                                      ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:448:47: error: ‘logLiquidFugacityDerivs’ cannot be used as a function
  448 |         real64 const dPhiLdx = logLiquidFugacityDerivs( ic, Deriv::dC+jc );
      |                        ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:449:47: error: ‘logVapourFugacityDerivs’ cannot be used as a function
  449 |         real64 const dPhiVdy = logVapourFugacityDerivs( ic, Deriv::dC+jc );
      |                        ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:466:63: error: ‘logLiquidFugacityDerivs’ cannot be used as a function
  466 |       X( ic + numComps, Deriv::dP ) = phiL * logLiquidFugacityDerivs( ic, Deriv::dP ) +
      |                                        ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:466:113: error: ‘logVapourFugacityDerivs’ cannot be used as a function
  466 |       X( ic + numComps, Deriv::dP ) = phiL * logLiquidFugacityDerivs( ic, Deriv::dP ) +
      |                                                                                                                 ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:468:63: error: ‘logLiquidFugacityDerivs’ cannot be used as a function
  468 |       X( ic + numComps, Deriv::dT ) = phiL * logLiquidFugacityDerivs( ic, Deriv::dT ) +
      |                                        ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:468:113: error: ‘logVapourFugacityDerivs’ cannot be used as a function
  468 |       X( ic + numComps, Deriv::dT ) = phiL * logLiquidFugacityDerivs( ic, Deriv::dT ) +
      |                                                                                                                 ^
make[3]: *** [coreComponents/constitutive/CMakeFiles/constitutive.dir/build.make:737: coreComponents/constitutive/CMakeFiles/constitutive.dir/fluid/multifluid/compositional/models/ImmiscibleWaterFlashModel.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:7742: coreComponents/constitutive/CMakeFiles/constitutive.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:2315: CMakeFiles/geosx.dir/rule] Error 2
make: *** [Makefile:348: geosx] Error 2
```

Second error:
```
[  9%] Building CUDA object coreComponents/constitutive/CMakeFiles/constitutive.dir/fluid/multifluid/compositional/models/PhillipsBrineDensity.cpp.o
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/models/PhillipsBrineDensity.cpp: In static member function ‘static void geos::constitutive::compositional::PhillipsBrineDensity::calculateEosWaterMolarVolume(geos::arraySlice1d<const double>&, geos::arraySlice1d<const double>&, const geos::constitutive::compositional::ComponentProperties&, geos::constitutive::compositional::EquationOfStateType, geos::real64, geos::integer, geos::arraySlice1d<double>&)’:
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/models/PhillipsBrineDensity.cpp:236:80: error: template argument 2 is invalid
  236 |   stackArray1d< real64, 2 + MultiFluidConstants::MAX_NUM_COMPONENTS > tempDerivs( numDofs );
      |                                                                                ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/models/PhillipsBrineDensity.cpp:273:213: error: request for member ‘toSlice’ in ‘tempDerivs’, which is of non-class type ‘int’
  273 |         CompositionalDensityUpdate::computeCompressibilityFactor( numComps,
      |                                                                                                                                                                                                                     ^
```

Third error:
```
[  9%] Building CUDA object coreComponents/constitutive/CMakeFiles/constitutive.dir/fluid/multifluid/compositional/CompositionalMultiphaseFluid.cpp.o
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp: In static member function ‘static void geos::constitutive::compositional::CubicEOSPhaseModel<EOS_TYPE>::computeCompressibilityFactor(geos::integer, geos::arraySlice1d<const double, USD>&, geos::arraySlice2d<const double>&, geos::arraySlice1d<const double>&, geos::arraySlice1d<const double>&, const geos::real64&, const geos::real64&, geos::real64&)’:
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp:877:69: error: template argument 2 is invalid
  877 |     stackArray1d< real64, MultiFluidConstants::MAX_NUM_COMPONENTS > logFugacityCoefficientsMax( numComps );
      |                                                                     ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp:878:69: error: template argument 2 is invalid
  878 |     stackArray1d< real64, MultiFluidConstants::MAX_NUM_COMPONENTS > logFugacityCoefficientsMin( numComps );
      |                                                                     ^
```

Fourth error:
```
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp: In static member function ‘static void geos::constitutive::compositional::NegativeTwoPhaseFlash::computeDerivatives(geos::integer, geos::real64, geos::real64, geos::arraySlice1d<const double>&, const geos::constitutive::compositional::ComponentProperties::KernelWrapper&, const geos::constitutive::compositional::FlashData&, const geos::real64&, geos::arraySlice1d<const double, USD1>&, geos::arraySlice1d<const double, USD1>&, geos::arraySlice1d<double, USD2>&, geos::arraySlice2d<double, USD3>&, geos::arraySlice2d<double, USD3>&)’:
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:379:69: error: template argument 2 is invalid
  379 |     stackArray1d< real64, maxNumComps > logLiquidFugacity( numComps );
      |                                                                     ^
path-to-geos/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:380:69: error: template argument 2 is invalid
  380 |     stackArray1d< real64, maxNumComps > logVapourFugacity( numComps );
      |                                                                     ^
```

--- a/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp
+++ b/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp
@@ -874,8 +874,8 @@ computeCompressibilityFactor( integer const numComps,
       }
     }
 
-    stackArray1d< real64, MultiFluidConstants::MAX_NUM_COMPONENTS > logFugacityCoefficientsMax( numComps );
-    stackArray1d< real64, MultiFluidConstants::MAX_NUM_COMPONENTS > logFugacityCoefficientsMin( numComps );
+    StackArray< real64, 1, MultiFluidConstants::MAX_NUM_COMPONENTS > logFugacityCoefficientsMax( numComps );
+    StackArray< real64, 1, MultiFluidConstants::MAX_NUM_COMPONENTS > logFugacityCoefficientsMin( numComps );
     computeLogFugacityCoefficients( numComps, composition, binaryInteractionCoefficients, zMin,
                                     aPureCoefficient, bPureCoefficient, aMixtureCoefficient, bMixtureCoefficient,
                                     logFugacityCoefficientsMin.toSlice() );

--- a/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp
+++ b/src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp
@@ -376,10 +376,10 @@ void NegativeTwoPhaseFlash::computeDerivatives(
   else
   {
     // Calculate the liquid and vapour fugacities and derivatives
-    stackArray1d< real64, maxNumComps > logLiquidFugacity( numComps );
-    stackArray1d< real64, maxNumComps > logVapourFugacity( numComps );
-    stackArray2d< real64, maxNumComps * maxNumDofs > logLiquidFugacityDerivs( numComps, numDofs );
-    stackArray2d< real64, maxNumComps * maxNumDofs > logVapourFugacityDerivs( numComps, numDofs );
+    StackArray< real64, 1, maxNumComps > logLiquidFugacity( numComps );
+    StackArray< real64, 1, maxNumComps > logVapourFugacity( numComps );
+    StackArray< real64, 2, maxNumComps * maxNumDofs, MatrixLayout::ROW_MAJOR_PERM > logLiquidFugacityDerivs( numComps, numDofs );
+    StackArray< real64, 2, maxNumComps * maxNumDofs, MatrixLayout::ROW_MAJOR_PERM > logVapourFugacityDerivs( numComps, numDofs );
 
     FugacityCalculator::computeLogFugacity( numComps,
                                             pressure,

--- a/src/coreComponents/constitutive/fluid/multifluid/compositional/models/ImmiscibleWaterFlashModel.hpp
+++ b/src/coreComponents/constitutive/fluid/multifluid/compositional/models/ImmiscibleWaterFlashModel.hpp
@@ -172,7 +172,7 @@ void ImmiscibleWaterFlashModelUpdate::compute( ComponentProperties::KernelWrappe
     // Hydrocarbon phases
 
     // Calculate normalised hyrdocarbon composition
-    stackArray1d< real64, maxNumComps > composition( m_numComponents );
+    StackArray< real64, 1, maxNumComps > composition( m_numComponents );
     for( integer ic = 0; ic < m_numComponents; ++ic )
     {
       composition[ic] = compFraction[ic] / z_hc;

--- a/src/coreComponents/constitutive/fluid/multifluid/compositional/models/PhillipsBrineDensity.cpp
+++ b/src/coreComponents/constitutive/fluid/multifluid/compositional/models/PhillipsBrineDensity.cpp
@@ -232,8 +232,8 @@ void PhillipsBrineDensity::calculateEosWaterMolarVolume( arraySlice1d< real64 co
   integer const numDofs = 2 + numComps;
 
   // Create pure water composition
-  stackArray1d< real64, MultiFluidConstants::MAX_NUM_COMPONENTS > waterComposition( numComps );
-  stackArray1d< real64, 2 + MultiFluidConstants::MAX_NUM_COMPONENTS > tempDerivs( numDofs );
+  StackArray< real64, 1, MultiFluidConstants::MAX_NUM_COMPONENTS > waterComposition( numComps );
+  StackArray< real64, 1, 2 + MultiFluidConstants::MAX_NUM_COMPONENTS > tempDerivs( numDofs );
   for( integer ic = 0; ic < numComps; ++ic )
   {
     waterComposition[ic] = 0.0;

